### ISSUE TITLE
revert handling

### ISSUE THREAD

**Issue by travi** [semantic-release/semantic-release#2959]
tl;dr: add an integration test for the revert behavior when a change is committed and reverted in the scope of the same release

the recent changes with the conventional-changelog dependencies [changed the order that the commits are processed](https://github.com/conventional-changelog/conventional-changelog/issues/1121#issuecomment-1709206347). we have tests in the related plugins to cover the expected revert behavior, but this change resulted in a need to change the test data. that sent me on an investigation to confirm that the real commit data would actually match the new order of the test data and i failed to find a higher level test that covered the scope of reading the commits and handling the revert behavior.

we should add an integration test in this project for this case since it would have a large enough scope to cover all of the pieces involved. in addition, i think there is an opportunity to extract some helper functions to reduce some duplication in the integration test file and make the tests easier to read and follow.
